api/envoy/api/v2/auth/cert.proto

```diff
@@ -315,6 +325,8 @@ message CommonTlsContext {
   repeated string alpn_protocols = 4;
 
   reserved 5;
+
+  PrivateKeyOperations private_key_operations = 9;
```

Comment:
This seems effectively a `oneof` with `TlsCertificate.private_key`. We can't promote it to a `oneof` without breaking the API, but should this field live next to that in this file and have a comment indicating the exclusive nature of these fields?

Reply:
We can't do `oneof` for wire compat reasons anyway, so you have flexibility here.

Reply:
I was under the impression that you can add `oneof` to non-repeatable fields and stay wire-compatible?

Reply:
Wire compatible yes, but it breaks Go language bindings. We've had a request from the gRPC team to avoid doing this gratuitously, this seems a reasonable accommodation to make. See https://github.com/envoyproxy/envoy/issues/6271.

Reply:
Ah, I wasn't aware of that, thanks! Yes, it makes sense to accommodate them (though, it's unfortunate that language bindings are source of this restriction).

Reply:
Ok, I'll move the private key operations provider to TlsCertificate. I'll have to set up the SSL socket bit differentlythen (because I'll need to initialize the private key methods for each `SSL_CTX`).